docs/CONNECTOR_PROTOCOL.md

```diff
@@ -97,16 +92,19 @@ This is our main communication index, used to communicate the connector's config
   last_sync_error: string;   -> Optional last job error message
   last_sync_status: string;  -> Status of the last job, or null if no job has been executed
   last_synced: date;    -> Date/time of last job (UTC)
-  name: string; -> the name to use for the connector
+  name: string;         -> the name to use for the connector
   pipeline: {
-    extract_binary_content: boolean; -> Whether the `request_pipeline` should handle binary data
-    name: string; ->  Ingest pipeline to utilize on indexing data to Elasticsearch
-    reduce_whitespace: boolean; -> Whether the `request_pipeline` should squish redundant whitespace
-    run_ml_inference: boolean; -> Whether the `request_pipeline` should run the ML Inference pipeline
-  }
+    extract_binary_content: boolean;    -> Whether the `request_pipeline` should handle binary data
+    name: string;                       ->  Ingest pipeline to utilize on indexing data to Elasticsearch
+    reduce_whitespace: boolean;         -> Whether the `request_pipeline` should squish redundant whitespace
+    run_ml_inference: boolean;          -> Whether the `request_pipeline` should run the ML Inference pipeline
+  };
+  preferences: {
+    extract_full_html: boolean; -> Whether or not to fully extract HTML (Crawler)
+  };                            -> Contains dynamic field:value pairs for preferences for the connector
```

Comment:
I think I've missed the discussion - why cannot it be a configurable field?

Reply:
@artem-shelkovnikov I think the big difference is a configuration is necessary to run, but this is just an opt-in to fetch a specific data type.

@sphilipse I remember discussing this with you but my notes just say "don't use configuration" -- do you remember why we decided that?

Reply:
Since you initialise and use the configuration, you can do the optional/required on your side of course, I just feel that having two objects that represent same concepts (e.g. alteration of behaviour of individual connector) might bite us back

Reply:
Because configuration fields show up in the UI, and are meant for service-type specific user-set configurations (like, say, username and password). This setting is independent from the service type, and shows up in a completely different part of the UI.

Reply:
Actually not independent from the service type, I guess, 'cause this'll only be used by crawlers. But still: as a type of configuration this shows up in a very different place, and conceptually is different from the configuration fields (which are more "how do I connect to my data source").

Reply:
The counterpoint to this would be that we don't actually use the configuration blob for the crawler, and the crawler already has a bunch of bespoke UIs, so maybe it's okay to put it in there anyway 🤔

Reply:
We do actually use the configuration blob for crawler. We add `{ "nextSyncConfig": { "label": "nextSyncConfig", "value": null } }` after running a crawl. In theory you could add configuration for the next crawl in place of `null` but we aren't using it in practice, it's a holdover from when we were trying to make crawler a connector.

Which leads me to a thought -- we don't really use `configuration` for crawler now, but if it became a connector in the future, we would. I also think there may be use cases for `preferences` on other connectors. For example if your DB has JSON objects in it, whether to store that in a document as JSON or string.

Reply:
> The counterpoint to this would be that we don't actually use the configuration blob for the crawler, and the crawler already has a bunch of bespoke UIs, so maybe it's okay to put it in there anyway 🤔

That was my feeling - we already have configurable_fields + features on the connectors side, but crawler is a very special connector with special UI that's not supported and won't be probably supported in generic connectors. I'm feeling like we should be really careful in this particular case.

Reply:
> won't be probably supported in generic connectors

This is true, Crawler will always be an awkward half-connector. It invalidates my point.
If we don't have a need for `preferences` for other connectors then maybe `configuration` is the way to go.